### PR TITLE
Add tests for `generate_uuid`, `of`, `truthy`, and `falsy` simul_efuns

### PR DESCRIPTION
**Pull request by gesslar**
Adds a test suite for the `util` simul_efuns covering `generate_uuid`, `of`, `truthy`, and `falsy`.

`generate_uuid` tests verify the v4 UUID format via regex, the expected 36-character length, and that successive calls produce distinct values.

`of` tests cover membership checks across arrays, mappings, and strings, including edge cases such as empty arrays, keys with a zero value, and non-container haystacks.

`truthy` and `falsy` tests confirm expected behavior for integers, empty and non-empty strings, and empty arrays (which are neither `0` nor `""` and therefore truthy).

<details><summary><h3>Greptile Summary</h3></summary>

This PR adds coverage for utility simul_efuns. The main changes are:
- UUID format, length, and uniqueness tests.
- Array, mapping, string, and non-container membership tests.
- Truthy and falsy predicate tests.
</details>

<sub>Reviews (2): Last reviewed commit: ["tests"](https://github.com/gesslar/oxidus-mudlib/commit/cc2080e5537b4064bf02d4c66ae3c4ee4c588de9) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=43716229)</sub>

<details><summary><h4>Context used (5)</h4></summary>

- Rule used - What: Commented-out code may remain only when it d... ([source](https://app.greptile.com/gesslar/-/custom-context?memory=a0d38c94-cefe-402e-b354-4a9a427afad4))
- Rule used - What: Release* and Quality* workflows must target ... ([source](https://app.greptile.com/gesslar/-/custom-context?memory=c8d8f233-9ea9-4bdc-8096-9102db58bf5b))
- Rule used - weighted priority system than just "P1, that thing... ([source](https://app.greptile.com/gesslar/-/custom-context?memory=0e553e6b-dbb2-4604-8a43-503f988cc3fd))
- Rule used - Do not flag cross-kind cache contamination in this... ([source](https://app.greptile.com/gesslar/github/gesslar/oxidus-mudlib/-/custom-context?memory=7f1b23a3-84a2-42dc-aead-132350aed221))
- Context used - This repository is Oxidus, a MUD library written i... ([source](.greptile))
</details>

### CODE REVIEW

**Comment by gesslar** [comment]
* **#299** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/299?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#298** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/298?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#297** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/297?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#296** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/296?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/296?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#295** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/295?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#294** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/294?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#293** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/293?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#292** <a href="https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/292?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: tests/adm/simul_efun/util.test.lpc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      ASSERT_EQ(0, of(1, ({})));
+    }),
+  }));
+
+  describe("of (mapping)", ({
+    test("finds a present key", function() {
+      ASSERT_EQ(1, of("key", ([ "key": 1 ])));
+    }),
+    test("rejects an absent key", function() {
+      ASSERT_EQ(0, of("missing", ([ "key": 1 ])));
```
[the remark this answers is not in the document]
Comment (gesslar) [reply tests/adm/simul_efun/util.test.lpc:51]: All my tests were green?

[the remark this answers is not in the document]
Comment (gesslar) [reply tests/adm/simul_efun/util.test.lpc:51]: oh, wait, nullp() is the same as undefinedp(). were you thinking they were different?

**Comment by gesslar** [comment]
### Merge activity

* **Jul 13, 3:00 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/296).
* **Jul 13, 3:08 AM UTC**: [Graphite](https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/296) rebased this pull request as part of a merge.
* **Jul 13, 3:08 AM UTC**: @gesslar merged this pull request with [Graphite](https://app.graphite.com/github/pr/gesslar/oxidus-mudlib/296).
